### PR TITLE
Merge upstream geth v1.13.9 into op-geth

### PR DESCRIPTION
**Pull request by roberto-bayardo**
**Description**

This brings geth commits up to v1.13.9 into op-geth.  The most notable change is the removal of the old simulated backend class and replacing it with a new one that rather unfortunately has more asynchrony in its behavior: If you call SendTransaction, then it is no longer guaranteed the transaction is in the "Pending" block at the point you call Commit(). This is a bit tricky to deal with during monorepo e2e testing (see in-progress monorepo PR linked below).

**Tests**

**Additional context**

Re: https://github.com/ethereum-optimism/protocol-quest/issues/186

Corresponding monorepo updates: https://github.com/ethereum-optimism/optimism/pull/9935

Some of this change was upstreamed here: https://github.com/ethereum/go-ethereum/pull/29316

**Metadata**

### CODE REVIEW

File: ethclient/simulated/backend.go [hunk 1]
```diff
@@ -145,10 +166,16 @@ func (n *Backend) Close() error {
 		n.client = simClient{}
 	}
 	if n.beacon != nil {
-		err := n.beacon.Stop()
+		_ = n.beacon.Stop()
 		n.beacon = nil
+	}
+	// OPTIMISM
```
Comment (roberto-bayardo) [inline ethclient/simulated/backend.go:172]: I should probably upstream this change.

File: ethclient/simulated/backend.go [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+		_ = n.beacon.Stop()
+		n.beacon = nil
+	}
+	// OPTIMISM
+	if n.eth != nil {
+		err := n.eth.Stop()
+		n.eth = nil
+		return err
+	}
+	// END OPTIMISM
```
Comment (roberto-bayardo) [inline ethclient/simulated/backend.go:178]: I should probably upsream this change, looks like an oversight.

**Comment by BlocksOnAChain** [comment]
@roberto-bayardo  - I added @op-geth-maintainers for a potential review, depending of their availability, so we can have more eyes on it and try to get those reviews sooner for you.

**Review by trianglesphere — COMMENTED** [review]
FYI I did merge in from `optimism`. Could we also update the version specified in [forkdiff (fork.yaml)](https://github.com/ethereum-optimism/op-geth/blob/optimism/fork.yaml). I do not believe that we need to modify the content for `fork.yaml` because it appeared to have very minimal changes

**Comment by roberto-bayardo** [comment]
@trianglesphere @sebastianst figured out a clean wait to get rid of the sleep() calls in the monorepo so I think this is good to go.

File: fork.yaml [hunk 3]
```diff
@@ -5,7 +5,7 @@ footer: |
 base:
   name: go-ethereum
   url: https://github.com/ethereum/go-ethereum
-  hash: b20b4a71598481443d60b261d3e5dcb37f8a0d82  # v1.13.8
+  hash: dd938d103d41f5255a40f7441ad7f0ceae0f7b07  # v1.13.9
```
Comment (trianglesphere) [inline fork.yaml:8]: We also need to modify the section at line 245 around the simulated backed

Comment (roberto-bayardo) [reply fork.yaml:8]: done

**Review by trianglesphere — APPROVED** [review]

**Comment by roberto-bayardo** [comment]
apologies! It was a merge against upstream geth, but I think a later rebase against op-geth must have broken it. redoing this PR here: https://github.com/ethereum-optimism/op-geth/pull/284

**Pull request merged**